### PR TITLE
Make rayon optional

### PR DESCRIPTION
**Pull request by voidxnull**
Added a `par_iter` feature and removed `Sync` bounds where they are not needed.
I also formatted the code with rustfmt, but can remove the commit if you think this is unnecessary.

### CODE REVIEW

File: tests/lib.rs [hunk 1]
```diff
@@ -85,6 +87,7 @@ fn arenatree_iter() {
     assert_eq!(node_refs, vec![&arena[a], &arena[b], &arena[c], &arena[d]]);
 }
 
+#[cfg(feature = "par_iter")]
```
Comment (saschagrunert) [inline tests/lib.rs]: Can you please change the appveyor and travis config to also test with this feature?

File: src/lib.rs [hunk 2]
```diff
@@ -363,19 +378,21 @@ impl NodeId {
...
+    pub fn insert_before<T>(self, new_sibling: NodeId, arena: &mut Arena<T>) {
         new_sibling.detach(arena);
         let previous_sibling_opt;
         let parent_opt;
         {
-            let (self_borrow, new_sibling_borrow) =
-                arena
-                    .nodes
-                    .get_pair_mut(self.index, new_sibling.index, "Can not insert a node before itself");
+            let (self_borrow, new_sibling_borrow) = arena.nodes.get_pair_mut(
```
Comment (saschagrunert) [inline src/lib.rs]: I think the formatting is okay, I just formatted it with 120 chars per line in my rustfmt config.

File: Cargo.toml [hunk 3]
```diff
@@ -17,8 +17,13 @@ appveyor = { repository = "saschagrunert/indextree", branch = "master", service
 
 [features]
 deser = [ "serde", "serde_derive" ]
+par_iter = ["rayon"]
```
Comment (saschagrunert) [inline Cargo.toml]: Can you please add a build including this feature in the appveyor and travis config?

**Comment by voidxnull** [comment]
Done. Removed the rustfmt commit and added the feature to the CI configs.

**Pull request merged**